modules/friendlytag.js

```diff
@@ -183,6 +183,248 @@ Twinkle.tag.callback = function friendlytagCallback() {
 		case 'redirect':
 			Window.setTitle('Redirect tagging');
 
+			// Certain redirect tags shouldn't be used in certain namespaces
+			// I generated the below lists using this search, and also added some manually.
+			// https://en.wikipedia.org/w/index.php?title=Special:Search&limit=500&offset=0&ns0=1&ns1=1&ns3=1&ns4=1&ns5=1&ns6=1&ns7=1&ns8=1&ns9=1&ns10=1&ns11=1&ns12=1&ns13=1&ns14=1&ns15=1&ns100=1&ns101=1&ns118=1&ns119=1&ns710=1&ns711=1&ns828=1&ns829=1&ns2300=1&ns2301=1&ns2302=1&ns2303=1&search=insource%3A%22Use+this+rcat+template+on%22
+			var mainspaceOnly = [
```

Comment:
All configurational data should be in`Twinkle.tag.redirectList`. You can add namespace filtering there, such as with `namespaces: [0, 1]` within the objects (with unspecified indicating all namespaces).

This avoids having to change things in multiple places while adding/removing tags to twinkle.

Reply:
Instead of doing `disallowedNamespaces: [1,2]` or something similar, I decided to do human readable statements, since one of the restrictions is not namespace related, and so that we wouldn't have to list out a bunch of numbers. The human readable restrictions are `restriction: insideMainspaceOnly, insideTalkNamespaceOnly, disambiguationPagesOnly, outsideUserspaceOnly`.